packages/next/src/build/swc/options.ts

```diff
@@ -482,6 +482,12 @@ export function getLoaderSWCOptions({
     options.isPageFile = false
     options.optimizeServerReact = undefined
     options.cjsRequireOptimizer = undefined
+    // Disable optimizer for node_modules in app browser layer, to avoid unnecessary replacement.
+    // e.g. typeof window could result differently in js worker or browser.
+    // override for non-jest mode
+    if (options.jsc.transform.optimizer.globals) {
```

Comment:
```suggestion
    if (options.jsc.transform.optimizer.globals?.typeofs) {
```
Should we make this check more specific